### ISSUE TITLE
Need of parameter v in Attention(Bahdanau) model?

### ISSUE THREAD

**Issue by kushalj001** [bentrevett/pytorch-seq2seq#74]
![image](https://user-images.githubusercontent.com/32245327/73563607-c08b9880-4483-11ea-9d86-11d68ee9a69e.png)  

Here, the notebook uses a parameter v and trains it via backpropagation. However, I am not able to find a similar parameter mentioned in the original paper. What is the exact function of v?   
![image](https://user-images.githubusercontent.com/32245327/73563760-24ae5c80-4484-11ea-9729-95811cdcbfc0.png)  
Here, in the paper, they have directly calculated the softmax for the _energies_ calculated in the notebook. The only function of v that I can get is to ensure that the size of attention is over the source length. If that's all, then what is the need for it to be trained?
Thank you.

**Comment by bentrevett** [comment]
Check the equation in the appendix section A.2.2, just under table 2.
